x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/SearchableSnapshotDirectory.java

```diff
@@ -361,68 +367,95 @@ private void prewarmCache() {
             final Executor executor = threadPool.executor(SEARCHABLE_SNAPSHOTS_THREAD_POOL_NAME);
             logger.debug("{} warming shard cache for [{}] files", shardId, cacheFiles.size());
 
+            final BlockingQueue<CheckedRunnable<Exception>> queue = new SizeBlockingQueue<>(new LinkedBlockingQueue<>(), Integer.MAX_VALUE);
```

Comment:
Mostly to prevent the queue to have more than Integer.MAX_VALUE elements, which could maybe happen for a large number of files with very low chunk size?